plugins/automap/src/automap.plugin.ts

```diff
@@ -0,0 +1,34 @@
+import { FlatfileEvent, FlatfileListener } from "@flatfile/listener";
+import { AutomapService } from "./automap.service";
+
+/**
+ * Automap plugin for Flatfile.
+ *
+ * @param options - config options
+ */
+export function automap(options: AutomapOptions) {
+  const automapper = new AutomapService(options);
+
+  return (listener: FlatfileListener): void => {
+    automapper.assignListeners(listener);
+  };
+}
+
+/**
+ * Plugin config options.
+ *
+ * @property {string} accuracy - match columns either by 'confident' (> 90% match) or 'exact' (100% match).
+ * @property {boolean} debug - show helpul messages useful for debugging (use intended for development).
+ * @property {string} defaultTargetSheet - exact sheet name to import data to.
+ * @property {RegExp} matchFilename - a regular expression to match specific files to perform automapping on.
+ * @property {Function} onFailure - callback to be executed when plugin bails.
+ * @property {string} targetWorkbook - specify destination Workbook id or name.
+ */
+export interface AutomapOptions {
+  readonly accuracy: "confident" | "exact";
```

Comment:
Chose 'confident' over 'remembered' b/c technically 'exact' is 'remembered' too. But I'm open to whatever